Sources/Datadog/Datadog.swift

```diff
@@ -205,13 +205,19 @@ public class Datadog {
         )
 
         // First, initialize features:
-        var rum: RUMFeature?
-
         var urlSessionAutoInstrumentation: URLSessionAutoInstrumentation?
-        var rumInstrumentation: RUMInstrumentation?
 
         var telemetry: RUMTelemetry?
 
+        if let urlSessionAutoInstrumentationConfiguration = configuration.urlSessionAutoInstrumentation {
+            urlSessionAutoInstrumentation = URLSessionAutoInstrumentation(
+                configuration: urlSessionAutoInstrumentationConfiguration,
+                dateProvider: configuration.common.dateProvider
+            )
+
+            core.register(feature: urlSessionAutoInstrumentation)
+        }
```

Comment:
This has been moved up so it's registered before RUM.
Please note that `URLSessionAutoInstrumentation` will be removed in a nextPR.